grpcwebclient/runner.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { Readable } from "node:stream";
+import puppeteer from "puppeteer";
+import * as esbuild from "esbuild";
+import {
+  ClientCompatRequest,
+  ClientCompatResponse,
+  ClientResponseResult,
+  ClientErrorResult,
+} from "./gen/proto/connectrpc/conformance/v1/client_compat_pb.js";
+
+export async function run() {
+  // Launch a browser. For a non-headless browser, pass false
+  const browser = await puppeteer.launch({ headless: "new" });
+  const page = await browser.newPage();
+
+  page.on("pageerror", (err) => {
```

Comment:
@timostamm I tried a few things to get this to trip, but I couldn't do it. I'm not sure if this is still needed since we have the try/catch below now?

Reply:
It's possible that it only triggers for uncaught errors, for example a syntax error in a script. I wouldn't worry about it too much with well-placed try-catch.